src/main/java/com/assignment/application/service/SalaryServiceImpl.java

```diff
@@ -93,32 +127,76 @@ public String updateSalary(Long companyId, SalaryUpdate salaryUpdate) {
                     double val = (salaryUpdate.getValue() / (double) 100) * salaryList.get(i).getSalary();
                     salaryList.get(i).setSalary(salaryList.get(i).getSalary() + val);
                 }
+            } else {
+                throw new InsufficientInformationException("Subtype not valid");
             }
             cachingInfo.updateSalary(salaryList, company.getId());
-        } else{
+        } else if (salaryUpdate.getType().equals("1")) {
             //dept of a company
-            Department department = departmentRepo.getDeptByCompId(companyId, salaryUpdate.getDepartmentName().toUpperCase());
-            if(department==null){
-                throw new RuntimeException("Department not exists");
+            Department department =
+                    null; //departmentRepo.getDeptByCompId(companyId, salaryUpdate.getDepartmentName().toUpperCase());
+            if (department == null) {
+                throw new NotExistsException("No such department exists");
             }
             List<Salary> salaryList = salaryRepo.salaryListCompDept(companyId, department.getId());
             if (salaryUpdate.getSubType().equals("0")) {
                 //update by amount
                 for (int i = 0; i < salaryList.size(); i++) {
                     salaryList.get(i).setSalary(salaryList.get(i).getSalary() + salaryUpdate.getValue());
                 }
-            } else {
+            } else if (salaryUpdate.getSubType().equals("1")) {
                 //update by percentage
                 for (int i = 0; i < salaryList.size(); i++) {
                     double val = (salaryUpdate.getValue() / (double) 100) * salaryList.get(i).getSalary();
                     salaryList.get(i).setSalary(salaryList.get(i).getSalary() + val);
                 }
+            } else {
+                throw new InsufficientInformationException("Subtype not valid");
             }
             cachingInfo.updateSalary(salaryList, company.getId());
+        } else {
+            throw new InsufficientInformationException("Type not valid");
         }
         kafkaTemplateSalary.send(SALARY_UPDATE_TOPIC, salaryUpdate);
         return StringConstant.UPDATE_SUCCESSFUL;
     }
 
-
+    @Override
+    public String updateSalaryOfEmployee(Long companyId, String employeeId, SalaryEmployeeUpdate salaryEmployeeUpdate) {
```

Comment:
Standard practice is to make method of max 20 lines. Avoid big methods. Create private functions for particular actions which can be used in multiple other functions. Instead of using setField, use Builder. All fields value can be set at once using Builder.